### ISSUE TITLE
'CosineAnnealingLR' never works with the wrong position of 'scheduler.step()'

### ISSUE THREAD

**Issue by GuohongLi** [sthalles/SimCLR#8]
Considering the setting in 'scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(optimizer, T_max=len(train_loader), eta_min=0, last_epoch=-1)'，I think 'scheduler.step()' should be called every step in 'for (xis,xjs),_ in train_loader'. Otherwise，lr will nerver change until 'len(train_loader)' epochs but not steps

**Comment by GuohongLi** [comment]
Mistake Fixs: "lr will nerver change"->"lr will nerver change over multiple cosine periods"

**Comment by sthalles** [comment]
Hello guys, that is a general message to say that I have refactored the whole project. I believe the project is much easier to understand now. Please have a look at the new impl and free to submit PR if you find any bugs. Thanks.
